### DISCUSSION TITLE
Why Teensy?

### DISCUSSION THREAD

**Discussion by ristiisa** [FundamentalFrequency/.github#67] [MCAD/ECAD]
I'm sorry if this is a dumb question, but why is Teensy needed for this project?
Could we not just wire the inputs directly to raspi4?

**Comment by stonepreston** [comment]
The Hyperpixel screen takes up all the GPIO pins on the pi.

**Comment by Pastitas** [comment]
It does not use them all however, as far as I've been able to find in their docs. It should also be able to use a arduino pro micro in place of the teensy (more availability, cheaper) for the purposes of sending midi messages. 
You could, theoretically, even use an arduino for the keys and encoders, and send  that data to the pi via I2C, this is done in other projects like zynthian.

**Reply by meirssv** [reply]
As I wrote in my post about the alternative build I am working on, ( #69) I use a Seeed Studio reTerminal. It has an integrated touchscreen (with even a higher resolution as the Hyperpixel-screen) and leaves the 40-pin header fully available. As I will also work with a Teensy and my own I2C RGB-encoders at first, the reTerminal offers the opportunity to indeed connect these I2C RGB-encoders directly to the Raspberry Pi CM4 module in the reTerminal and leaving out the Teensy in the end.
